unittests/main.cpp

```diff
@@ -2077,5 +2123,11 @@ TEST(Backend, CacheIntegration)
             std::cout << "checked" << std::endl;
         }
     }
-}
 
+    done = true;
+    work.reset();
+
+    });
+
+    ioc.run();
+}
```

Comment:
Not in this PR, but in the code anyway, when doing checkin(), in addition to clearing the results, the asio socket should be cancelled. This aborts any pending asio async operations.

    /** Cancel asynchronous operations on underlying postgres socket. */
    void
    cancel()
    {
        if (socket_)
            try { socket_->cancel(); } catch (...) {}
    }